src/pages/workspaces/workspace/analysis/file-utils.ts

```diff
@@ -47,6 +55,30 @@ export interface AnalysisFileStore {
   loading: boolean
 }
 
+export const useAnalysisFiles = (): AnalysisFileStore => {
+  const signal = useCancellation()
+  const [loading, setLoading] = useState(false)
+  const workspace: WorkspaceWrapper = useStore(workspaceStore)
+  const analyses: AnalysisFile[] = useStore(analysisFilesStore)
```

Comment:
Why does this need a global store? This adds complexity since now we have to deal with clearing this store when changing workspaces.

Reply:
hmmm... so I changed this to use `useEffect`, with `refresh` running again when the workspace store is updated. Let me know what you think

Reply:
That sounds like it's more reliable, but I still think the global store adds unnecessary complexity for no benefit.

Reply:
Hmm... as is it has no benefit. I was hoping that, down the line, we can make these IA stores and hooks global so as to have the option to 'cache' the info unless workspace state changes. Not going to include that here though since it would require further changes/considerations.